h/static/scripts/directive/excerpt.js

```diff
@@ -69,63 +69,126 @@ function excerpt() {
     controllerAs: 'vm',
     link: function (scope, elem, attrs, ctrl) {
```

Comment:
It seems to me that this link function is getting a bit unwieldy. Could you encapsulate the functionality of the overflow checking in a component that could be independently unit tested, rather than simply testing the functional behaviour of a fairly large and inscrutable blob of code?